eng/verify-nupkgs.ps1

```diff
@@ -112,4 +116,50 @@ function Unzip {
     [System.IO.Compression.ZipFile]::ExtractToDirectory($zipfile, $outpath)
 }
 
+function Match-VersionAgainstBranch {
+    param ([string]$vsTestVersion, [string]$branchName)
+
+    # Output useful info.
+    Write-Host "VSTest Product Version: `"$vsTestVersion`""
+    Write-Host "Current Branch: `"$branchName`""
+
+    $versionIsRTM = $vsTestVersion -match "^\d+\.\d+\.\d+$"
+    $versionIsRelease = $vsTestVersion -match "^\d+\.\d+\.\d+\-release\-\d{8}\-\d{2}$"
+    $versionIsPreview = $vsTestVersion -match "^\d+\.\d+\.\d+\-preview\-\d{8}\-\d{2}$"
+
+    $isReleaseBranch = $branchName -like "rel/*"
+    $isPreviewBranch = $branchName -like "main"
+
+    if (!$isReleaseBranch -and !$isPreviewBranch)
+    {
+        Write-Host "Skipping check since branch is neither `"release`" nor `"preview`""
+        return
+    }
+
+    Write-Host "Matching branch against product version ... "
+    if ($isReleaseBranch -and !$versionIsRTM -and !$versionIsRelease)
+    {
+        Write-Error "Release version `"$vsTestVersion`" should either be RTM, or contain a `"release`" suffix."
+        Exit 1
+    }
+    if ($isPreviewBranch -and !$versionIsPreview)
+    {
+        Write-Error "Preview version `"$vsTestVersion`" should contain a `"preview`" suffix."
+        Exit 2
```

Comment:
Append to the $errors collection and don't exit

Reply:
Fixed.